app/models/concerns/user_contest_scope.rb

```diff
@@ -17,7 +17,7 @@ module UserContestScope
         end
         .group(:id)
         .select('user_contests.id as id, sum(case when ' \
-        'short_submissions.answer = short_problems.answer then 1 else 0 end) ' \
+        'short_submissions.answer = short_problems.answer then short_problems.correct when short_submissions.answer = null then short_problems.empty else short_problems.wrong end) ' \
```

Comment:
You might also need to check for empty string (`""`).

Reply:
what the different between "" and null

Reply:
I'm actually not sure. I suspect that the client might be able to send an empty string to our servers and it'll be the empty string in the database. We still want to treat them as an empty answer though instead of a wrong one.